src/main/java/spoon/support/reflect/code/CtBlockImpl.java

```diff
@@ -42,15 +39,29 @@
 	private static final long serialVersionUID = 1L;
 
 	@MetamodelPropertyField(role = CtRole.STATEMENT)
-	private List<CtStatement> statements = emptyList();
+	private final List<CtStatement> statements = new ModelList<CtStatement>(this) {
+		@Override
+		protected CtRole getRole() {
+			return STATEMENT;
+		}
+		@Override
+		protected int getDefaultCapacity() {
+			return BLOCK_STATEMENTS_CONTAINER_DEFAULT_CAPACITY;
+		}
+		@Override
+		protected void onSizeChanged(int newSize) {
+			if (isImplicit() && (newSize > 1 || newSize == 0)) {
+				setImplicit(false);
+			}
+		}
+	};
 
 	public void accept(CtVisitor visitor) {
 		visitor.visitCtBlock(this);
 	}
 
 	@Override
 	public List<CtStatement> getStatements() {
-		ensureModifiableStatementsList();
 		return this.statements;
```

Comment:
Why does this method return a mutable list? It seems on purpose regarding the previous call of `ensureModifiableStatementsList` but it feels wrong...

Reply:
It returns a mutable ModelList, which can be mutated using List API and which assures that spoon model is kept consistent (parent-child is kept) and that modification events are sent.

This change is the CORE idea of this PR ;-)

Reply:
> This change is the CORE idea of this PR ;-)

Hmmm. It looks a bit dangerous for me: old users won't change their API and won't beneficiate the changes, but it might disturb new users because they won't necessarily expect the API to return a pointer on the model. 

WDYT about keeping a `getStatements` that returns an immutable list, and adding a `getModifiableStatements` like the one you proposed?

Reply:
Note: that this method already returned modifiable list before. I just assured that this modifiable list behaves correctly.

Reply:
> that this method already returned modifiable list before

My point is that I believe we should have the same consistent behaviour everywhere in Spoon. And I think that almost everywhere - but maybe not here - this kind of method is currently returning a new ArrayList, detached from the model  (see for example CtClass#getConstructors or CtType#getDeclaredExecutables): so they are mutable, but not linked to the model. If we make those methods returning a mutable element which is now linked to the model, it really can bring unexpected behaviours for the clients. 

That's why I suggest to start here to fix a definitive behaviour that we'll use everywhere. And I'll definitively prefer to change those methods to return an immutable object and to create another methods which return explicitely a mutable linked to the model: we might break some clients, but the fix is easy and explicit.